pages/index.vue

```diff
@@ -147,15 +188,56 @@ export default defineComponent({
      * @todo 1. Send the request to update the task to the backend server.
      * @todo 2. Update the task in the dom.
      */
-    updateTask(_index, _id) {},
+    /*async */updateTask(_index, _id) {
+
+      let newTitle = this.todos[_index].title;
+      let token = this.$store.getters.token;
+      //console.log(newTitle);
+      let all = this;
+
+      if(newTitle!= '') {
+          /*await*/ this.$axios({
+            url: 'https://todo-app-csoc.herokuapp.com/' + "todo/" + _id + "/",
+            headers: {
+                Authorization: "Token " + token,
+            },
+            method: "patch",
+            data: {
+                id: _id,
+                title: newTitle,
+            }
+        }).then(function({data, status}) {
+            all.$toast.success("done");
+            //console.log("success")
+        }).catch(function(err) {
+            all.$toast.error("something went wrong");
+            //console.log(err)
+        })
+      } else {
+        this.$toast.error("empty field");
+        this.$axios({
+        headers: {
+            Authorization: 'Token ' + token,
+        },
+        url: 'https://todo-app-csoc.herokuapp.com/' + 'todo/' +_id +'/',
+        method: 'get',
+      }).then(function({data, status}) { 
+        //console.log(data)
+        all.todos[_index].title = data.title;
```

Comment:
you could have used a separate v-model for input for all the tasks instead of calling the API again

Reply:
yeah that would have been a better method ig